httptrace/client.go

```diff
@@ -174,3 +175,43 @@ func cloneRequest(r *http.Request) *http.Request {
 	}
 	return r2
 }
+
+// saveCloneRequest keeps clone request for CancelRequest
+func (t *Transport) saveCloneRequest(original *http.Request, clone *http.Request) {
```

Comment:
I think the name `saveCloneRequest` is a bit misleading because it also deletes the saved clone request in the second call on line 161. How about `setCloneRequest` instead?

Also, could we make the comment instead read:

```
// setCloneRequest keeps track of the cloned based on the original request so
// that it can be canceled in the future in CancelRequest.
```